src/commands/PublishCommand.js

```diff
@@ -338,14 +338,15 @@ export default class PublishCommand extends Command {
     });
 
     this.progressBar.init(this.packagesToPublish.length);
+    const registry = this.getOptions().registry;
```

Comment:
Let's set this as an instance prop in the `initialize` method, like `this.npmRegistry = this.getOptions().registry;`